### PR TITLE
Feature/85 forgot password page

### LINKED ISSUE #85: Forgot password page android & ios

### PR DESCRIPTION
**Pull request by sebastian-vangrieken**
you can now request a password reset on android and ios

### CODE REVIEW

File: composeApp/src/androidMain/kotlin/com/wisemen/chargehub/ui/screens/ForgotPasswordScreen.kt [hunk 1]
```diff
@@ -0,0 +1,131 @@
...
+        },
+    ) { contentPadding ->
+        Column(
+            modifier = Modifier.padding(contentPadding).fillMaxWidth(),
+            horizontalAlignment = Alignment.CenterHorizontally
+        ) {
+            Text(
+                modifier = Modifier.padding(horizontal = Spacing.medium).padding(top = 70.dp, bottom = 30.dp),
+                textAlign = TextAlign.Center,
+                text = "Geef je e-mail adres in en volg de instructies om je wachtwoord te resetten.",
```
Comment (DanneD-Wisemen) [inline composeApp/src/androidMain/kotlin/com/wisemen/chargehub/ui/screens/ForgotPasswordScreen.kt]: Hardcoded text

File: shared/src/commonMain/kotlin/screens/forgotpassword/ForgotPasswordViewModel.kt [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+
+import com.rickclephas.kmm.viewmodel.coroutineScope
+import db.repository.FirebaseRepository
+import kotlinx.coroutines.launch
+import screens.AbstractViewModel
+
+class ForgotPasswordViewModel(private val firebaseRepo: FirebaseRepository) :
+    AbstractViewModel<ForgotPasswordUiAction, ForgotPasswordUiEvent, ForgotPasswordUiState>(ForgotPasswordUiState()) {
+        override fun onAction(action: ForgotPasswordUiAction) = viewModelScope.coroutineScope.launch {
+        when (action) {
```
Comment (DanneD-Wisemen) [inline shared/src/commonMain/kotlin/screens/forgotpassword/ForgotPasswordViewModel.kt:22]: watch indentiations

**Review by DanneD-Wisemen — CHANGES_REQUESTED** [review]

File: composeApp/src/androidMain/kotlin/com/wisemen/chargehub/ui/screens/ForgotPasswordScreen.kt [hunk 3]
```diff
@@ -0,0 +1,131 @@
...
+                        maxLines = 1,
+                        overflow = TextOverflow.Ellipsis,
+                        style = TextStyles.topBarTitle,
+                    )
+                },
+                navigationIcon = {
+                    IconButton(
+                        onClick = { onAction(ForgotPasswordUiAction.OnClickedBackAction) },
+                    ) {
+                        Row {
```
Comment (Jesse-Appwise) [inline composeApp/src/androidMain/kotlin/com/wisemen/chargehub/ui/screens/ForgotPasswordScreen.kt]: Why wrap this in a row?

File: iosApp/iosApp/ui/pages/ForgotPasswordScreen.swift [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+
+struct ForgotPasswordScreen: View {
+    @ObservedObject var navigation: NavigationController
+    @StateViewModel var viewModel = ForgotPasswordViewModel(firebaseRepo: FirebaseRepository())
+    let strings = StringsHelper().getResourceStrings()
+    
+    var body: some View {
+        HStack {
+            Button(action: {
+                navigation.navigateBack()
```
Comment (Jesse-Appwise) [inline iosApp/iosApp/ui/pages/ForgotPasswordScreen.swift]: Isn't it possible to use the same action/event based way as android?

Comment (Jesse-Appwise) [reply iosApp/iosApp/ui/pages/ForgotPasswordScreen.swift]: In this case the onAction(ForgotPasswordAction.OnNavigateBackClicked) and fetching the event to navigate back

**Review by Jesse-Appwise — CHANGES_REQUESTED** [review]

File: composeApp/src/androidMain/kotlin/com/wisemen/chargehub/ui/screens/ForgotPasswordScreen.kt [hunk 5]
```diff
@@ -0,0 +1,133 @@
...
+                isValid = true,
+                onInputChanged = { onAction(ForgotPasswordUiAction.OnEmailChangedAction(it)) },
+            )
+            Buttons.PrimaryButton(
+                modifier = Modifier.fillMaxWidth().padding(horizontal = Spacing.medium),
+                text = stringResource(R.string.send_mail),
+                colors = ButtonDefaults.primaryButtonColors(),
+                onClick = { onAction(ForgotPasswordUiAction.OnClickedSendEmailAction) },
+            )
+            Spacer(modifier = Modifier.weight(1f))
```
Comment (DanneD-Wisemen) [inline composeApp/src/androidMain/kotlin/com/wisemen/chargehub/ui/screens/ForgotPasswordScreen.kt]: Is this spacer necessary? Also add whitespacing between the composables

File: composeApp/src/main/res/values/strings.xml [hunk 6]
```diff
@@ -1,3 +1,65 @@
...
+    <string name="type_here">Type here…</string>
+    <string name="streamline_your_day_with">Stroomlijn je dag met</string>
+    <string name="welcome_back">Welkom Terug!</string>
+    <string name="login_welcome_text">Fijn om je terug te zien</string>
+    <string name="forgot_password">Wachtwoord vergeten?</string>
+    <string name="privacy_policy">Privacybeleid</string>
+    <string name="vehicle">Jouw Voertuig</string>
+    <string name="laadpaal">Laadpaal %1$s</string>
+    <string name="_28_september">28 September</string>
+    <string name="start">START</string>
```
Comment (DanneD-Wisemen) [inline composeApp/src/main/res/values/strings.xml:12]: Don't capitalize string resources. This should be done in the presentation layer (your view / composable)

File: iosApp/iosApp/ui/pages/ForgotPasswordScreen.swift [hunk 7]
```diff
@@ -0,0 +1,35 @@
+import SwiftUI
+import Shared
+import KMMViewModelSwiftUI
+
+struct ForgotPasswordScreen: View {
+    @StateViewModel var viewModel = ForgotPasswordViewModel(firebaseRepo: FirebaseRepository())
+    let strings = StringsHelper().getResourceStrings()
+    
+    var body: some View {
+        VStack {
```
Comment (DanneD-Wisemen) [inline iosApp/iosApp/ui/pages/ForgotPasswordScreen.swift:10]: Whitespacing between components

File: iosApp/iosApp/ui/pages/LoginScreen.swift [hunk 8]
```diff
@@ -37,7 +37,11 @@ struct LoginScreenView: View {
                         )
                     ).padding(.bottom, 4)
                     HStack {
-                        Text(strings.get(id: SharedRes.strings().forgot_password, args: [])).font(.system(size: 12)).fontWeight(.regular)
+                        NavigationLink {
+                            ForgotPasswordScreen()
+                        } label: {
+                            Text("Wachtwoord vergeten?").font(.system(size: 12)).fontWeight(.regular)
```
Comment (DanneD-Wisemen) [inline iosApp/iosApp/ui/pages/LoginScreen.swift:47]: Why did you remove using the string resource to make it hardcoded text?

**Review by DanneD-Wisemen — CHANGES_REQUESTED** [review]

File: iosApp/iosApp/ui/pages/LoginScreen.swift (hunk 8 above)
Comment (sebastian-vangrieken) [reply iosApp/iosApp/ui/pages/LoginScreen.swift:47]: I think I changed this in another branch but this branch is just not up to date yet.

**Review by DanneD-Wisemen — APPROVED** [review]
